tests/std/tests/GH_002769_handle_deque_block_pointers/test.cpp

```diff
@@ -328,8 +329,32 @@ void test_inconsistent_difference_types() {
     assert(counter == 0);
 }
 
+// Also test GH-4954: Endless loop in deque::shrink_to_fit()
+void test_gh_4954() {
+    deque<int> qu;
+
+    int it = 0;
+    while (it < 1024) {
+        const auto numAlloc = static_cast<size_t>(rand() + 1);

```

Comment:
Yeah, I'm using `mt19937_64` now instead. The count 1024 is reduced to 256 because the bug was triggered when `i == 74` and 1024 caused the test program take a bit long time on my old laptop.

Reply:
For future reference, `rand()` has been upgraded from "the STL maintainers will ugh and bleah and link to their decade-old presentations" to *forbidden by Microsoft policy* except when needed to implement the C and C++ Standards. We will never accept any additional use of `rand()` in the codebase.